src/pkcs11/misc.c

```diff
@@ -202,9 +201,11 @@ CK_RV push_login_state(struct sc_pkcs11_slot *slot,
 	r = CKR_OK;
 
 err:
-	if (r != CKR_OK && login) {
-		sc_mem_clear(login->pPin, login->ulPinLen);
-		free(login->pPin);
+	if (login) {
```

Comment:
There were two separate problems reported by Coverity in context of this chunk of code:

    Error: CLANG_WARNING: [#def125]
    opensc-0.16.0/src/pkcs11/misc.c:211:9: warning: Potential leak of memory pointed to by 'login'

and

    Error: CLANG_WARNING: [#def124]
    opensc-0.16.0/src/pkcs11/misc.c:206:3: warning: Function call argument is an uninitialized value

You are probably right about the first one. Coverity missed the case the `login` got stored in the `slot->logins`. On success we should not free the variable. Thanks for catching that (added in the new commit).

The second problem was when the `sc_mem_alloc_secure()` failed, we were trying to call `sc_mem_clear()` with undefined `login->ulPinLen` (therefore the second check).